src/core/Akka.Streams.Tests/Dsl/GraphMergePrioritizedSpec.cs

```diff
@@ -0,0 +1,180 @@
+            }
+
+            double ones = collected.Count(x => x == 1);
+            double twos = collected.Count(x => x == 2);
+            double threes = collected.Count(x => x == 3);
+
+            Math.Round(ones / twos).Should().Be(2);
+            Math.Round(ones / threes).Should().Be(6);
+            Math.Round(twos / threes).Should().Be(3);
+        }
+
+        [Fact]
+        public void MergePrioritized_must_stream_data_when_only_one_source_produces()
+        {
+            var elementCount = 10;
+            var source1 = Source.FromEnumerator(() => Vector.Fill<int>(elementCount)(() => 1).GetEnumerator());
+            var source2 = Source.FromEnumerator(() => new List<int>().GetEnumerator());
+            var source3 = Source.FromEnumerator(() => new List<int>().GetEnumerator());
+
+            var priorities = new List<int> { 6, 3, 1 };
+            var probe = this.CreateManualSubscriberProbe<int>();
+
+            ThreeSourceMerge(source1, source2, source3, priorities, probe).Run(Materializer);
+
+            var subscription = probe.ExpectSubscription();
+
+            var collected = new HashSet<int>();
+            for (int i = 1; i <= elementCount; i++)
+            {
+                subscription.Request(1);
+                collected.Add(probe.ExpectNext());
+            }
+
+            int ones = collected.Count(x => x == 1);
+            int twos = collected.Count(x => x == 2);
+            int threes = collected.Count(x => x == 3);
+
+            ones.Should().Be(elementCount);
+            twos.Should().Be(0);
+            threes.Should().Be(0);
+        }
+
+        [Fact]
+        public void MergePrioritized_must_stream_data_with_priority_when_only_two_sources_produce()
+        {
+            var elementCount = 20000;
+            var source1 = Source.FromEnumerator(() => Vector.Fill<int>(elementCount)(() => 1).GetEnumerator());
+            var source2 = Source.FromEnumerator(() => Vector.Fill<int>(elementCount)(() => 2).GetEnumerator());
+            var source3 = Source.FromEnumerator(() => new List<int>().GetEnumerator());
+
+            var priorities = new List<int> { 6, 3, 1 };
+            var probe = this.CreateManualSubscriberProbe<int>();
+
+            ThreeSourceMerge(source1, source2, source3, priorities, probe).Run(Materializer);
+
+            var subscription = probe.ExpectSubscription();
+
+            var collected = new HashSet<int>();
+            for (int i = 1; i <= elementCount; i++)
+            {
+                subscription.Request(1);
+                collected.Add(probe.ExpectNext());
+            }
+
+            double ones = collected.Count(x => x == 1);
+            double twos = collected.Count(x => x == 2);
+            int threes = collected.Count(x => x == 3);
+
+            threes.Should().Be(0);
+            Math.Round(ones / twos).Should().Be(2);
+        }
+
+        private RunnableGraph<T> ThreeSourceMerge<T>(Source<T, NotUsed> source1, Source<T, NotUsed> source2,
```

Comment:
@Horusiath @Silv3rcircl3  I'm not sure that I've converted it properly. The original source looks like
```scala
  private def threeSourceMerge[T](source1: Source[T, NotUsed], source2: Source[T, NotUsed], source3: Source[T, NotUsed], priorities: Seq[Int], probe: ManualProbe[T]) = {
    RunnableGraph.fromGraph(GraphDSL.create(source1, source2, source3)((_, _, _)) { implicit b ⇒ (s1, s2, s3) ⇒
      val merge = b.add(MergePrioritized[T](priorities))
      s1.out ~> merge.in(0)
      s2.out ~> merge.in(1)
      s3.out ~> merge.in(2)
      merge.out ~> Sink.fromSubscriber(probe)
      ClosedShape
    })
  }
```